org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -108,6 +108,9 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
     logger.setUseParentHandlers(false);
     Properties.setConnectionId(connectionId);
 
+    MetricClient metricClient = new AWSMetricClient(context.getFunctionName());
```

Comment:
I think the FunctionName will be something long and different for each deployment of Javabuilder? Something like `javabuilder-[subdomain]-BuildAndRunJavaConsoleProjectFun-[randomid]`

If it’s not too much extra work, it might be useful to just extract the logical name of the Function (`BuildAndRunJava`), the Project Sub-Type (`ConsoleProject`), and the Stack Name / Subdomain (`beta`, `integration`, `production`, etc.), and publish each of those as separate Dimensions of the Metric.

That may make it easier for us to track Metrics by a given instance of Javabuilder, or by a given function type.

Reply:
I would rather not manually parse the function name because that is fragile if we ever decide to change the pattern. I like the idea of splitting it up, but in order to do that we would need to send a few more pieces of information to each lambda (and still probably wouldn't have the stack name unless we did that manually parsing). For what it's worth the default metrics you get on a lambda have the dimension of function name, so we are following an established pattern.